lib/tbot/service_workload_identity_aws_ra_test.go

```diff
@@ -0,0 +1,51 @@
+// Teleport
+// Copyright (C) 2025 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+package tbot
+
+import (
+	"context"
+	"testing"
+
+	"github.com/spiffe/aws-spiffe-workload-helper/vendoredaws"
+	"github.com/stretchr/testify/require"
+
+	"github.com/gravitational/teleport/lib/tbot/config"
+	"github.com/gravitational/teleport/lib/utils/testutils/golden"
+)
+
+func Test_renderAWSCreds(t *testing.T) {
```

Comment:
It'd be great to have some test cases that exercise the actual SVID → AWS credentials exchange. Even if that means mocking out the AWS SDK methods. What do you think?

Reply:
Gave this a shot in [f5d4768](https://github.com/gravitational/teleport/pull/52426/commits/f5d476889f9e1c8877681e9560f930da827fed4c) - I didn't manage to test the actual signature produced but that comes from AWS code which has their own tests over it. I think this is close enough without introducing additional complexity.